src/components/PDFThumbnail/index.tsx

```diff
@@ -0,0 +1,51 @@
+// @ts-expect-error - We use the same method as PDFView to import the worker
+import pdfWorkerSource from 'pdfjs-dist/legacy/build/pdf.worker';
+import React, {useEffect, useMemo} from 'react';
+import {View} from 'react-native';
+import {Document, pdfjs, Thumbnail} from 'react-pdf';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import useThemeStyles from '@hooks/useThemeStyles';
+import addEncryptedAuthTokenToURL from '@libs/addEncryptedAuthTokenToURL';
+import type PDFThumbnailProps from './types';
+
+function PDFThumbnail({previewSourceURL, style, isAuthTokenRequired = false, enabled = true, onPassword = () => {}}: PDFThumbnailProps) {
+    const styles = useThemeStyles();
+
+    useEffect(() => {
+        const workerURL = URL.createObjectURL(new Blob([pdfWorkerSource], {type: 'text/javascript'}));
+        if (pdfjs.GlobalWorkerOptions.workerSrc !== workerURL) {
+            pdfjs.GlobalWorkerOptions.workerSrc = workerURL;
+        }
+    }, []);
```

Comment:
The useEffect will execute only after render, during first render the pdf worker may not work as expected. Move outside useEffect.